functional/exec_test.go

```diff
@@ -13,3 +13,104 @@
 // limitations under the License.
 
 package functional
+
+import (
+	"fmt"
+	"time"
+
+	. "github.com/clearcontainers/tests"
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/ginkgo/extensions/table"
+	. "github.com/onsi/gomega"
+)
+
+var sleepingContainerWorkload = []string{"sh", "-c", "sleep 1000"}
+
+func execDetachTiming(detach bool) TableEntry {
+	comparator := ">="
+	if detach {
+		comparator = "<"
+	}
+
+	timeout := 1
+
+	process := Process{
+		Workload: []string{"/bin/sleep", fmt.Sprintf("%d", timeout)},
+		Detach:   detach,
+	}
+
+	return Entry(fmt.Sprintf("check time as detach=%t", detach), process, comparator, timeout)
+}
+
+func execDetachOutput(detach bool) TableEntry {
+	outputHelloWorld := "HelloWorld"
+
+	process := Process{
+		Workload: []string{"echo", outputHelloWorld},
+		Detach:   detach,
+	}
+
+	expectedOutput := outputHelloWorld
+	if detach {
+		expectedOutput = ""
+	}
+
+	return Entry(fmt.Sprintf("check output as detach=%t", detach), process, expectedOutput)
+}
+
+var _ = Describe("exec", func() {
+	var (
+		container *Container
+		exitCode  int
+		err       error
+	)
+
+	BeforeEach(func() {
+		container, err = NewContainer(sleepingContainerWorkload, true)
+		Ω(container).ShouldNot(BeNil())
```

Comment:
@grahamwhaley  according with the documentation http://onsi.github.io/gomega/#making-assertions
```Ω```  is ok, because He is using ```ShouldNot```